pkg/resiliency/noop_test.go

```diff
@@ -0,0 +1,55 @@
+package resiliency
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNoOp(t *testing.T) {
```

Comment:
Should the no-op policy have tests that show it is truly a no-op (return failure and not retry).

Reply:
Yeah I'll tweak the tests to return an error for each call type.